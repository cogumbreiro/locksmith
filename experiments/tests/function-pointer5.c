#include <pthread.h>
#include <stdio.h>

typedef void (*fptr)(void*);

int i;

void bar(void* arg) {
  fptr x = (fptr) arg;
  i++;//race
  x(arg);
}

fptr foo = &bar;

int main() {
  pthread_t t1;
  pthread_create(&t1, NULL, foo, foo);
  foo(foo);
}


