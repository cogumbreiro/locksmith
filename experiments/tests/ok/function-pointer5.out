************************* STARTING *************************

shared: &i:function-pointer5.c:6

Warning: Possible data race: &i:function-pointer5.c:6 is not protected!
 references:
  dereference of &i:function-pointer5.c:6 at function-pointer5.c:10
    &i:function-pointer5.c:6
  locks acquired:
    <empty>
  in: FORK at function-pointer5.c:18 
             

  dereference of &i:function-pointer5.c:6 at function-pointer5.c:10
    &i:function-pointer5.c:6
  locks acquired:
    <empty>
  in: FORK at function-pointer5.c:18  -> function-pointer5.c:11
             

  dereference of &i:function-pointer5.c:6 at function-pointer5.c:10
    &i:function-pointer5.c:6
  locks acquired:
    <empty>
  in: main at function-pointer5.c:16 
             

  dereference of &i:function-pointer5.c:6 at function-pointer5.c:10
    &i:function-pointer5.c:6
  locks acquired:
    <empty>
  in: main at function-pointer5.c:16  -> function-pointer5.c:19
             

  dereference of &i:function-pointer5.c:6 at function-pointer5.c:10
    &i:function-pointer5.c:6
  locks acquired:
    <empty>
  in: main at function-pointer5.c:16  -> function-pointer5.c:11
             

  dereference of &i:function-pointer5.c:6 at function-pointer5.c:10
    &i:function-pointer5.c:6
  locks acquired:
    <empty>
  in: main at function-pointer5.c:16  -> function-pointer5.c:19
                          

  dereference of &i:function-pointer5.c:6 at function-pointer5.c:10
    &i:function-pointer5.c:6
  locks acquired:
    <empty>
  in: main at function-pointer5.c:16  -> function-pointer5.c:19 -> function-pointer5.c:11
             

*************************** DONE ***************************
